code/script.js

```diff
@@ -240,104 +420,166 @@ const generateBoard = () => {
+  generateBoard();
+  setSecret();
+};
 
-// setting the currentQuestion object when you select something in the dropdown
+// setting the currentQuestion object when you select something in the dropdown and defines which optground it is connected to.
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.value;
 
-  if (category === 'hair color') {
+  if (category === "hair color") {
     currentQuestion = {
-      attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      attribute: "hairColor",
+      value: value,
       category: category,
-    }
-  } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
+    };
+  } else if (category === "eye color") {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: "eyeColor",
+      value: value,
       category: category,
-    }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    };
+  } else if (category === "accessories") {
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    };
+  } else if (category === "other") {
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    };
   }
-}
+};
 
-// This function should be invoked when you click on 'Find Out'.
+//Filters after comparing the value of the current question with the secret person.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-}
+  const keep = currentQuestion.value === secret[currentQuestion.attribute];
+  filterCharacters(keep);
+};
 
-// It'll filter the characters array and redraw the game board.
+// Triggers an alert telling whether the secret person has the attribute chosen or not, and redraws a filtered board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const {category, attribute, value} = currentQuestion
+  if (category === "accessories") {
     if (keep) {
       alert(
         `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[attribute] === value
+      );
```

Comment:
I know that there are ways to shorten the code by writing the filtering for all the characters. But, as a beginner I find it very easy to understand the flow of the game with your way of writing it. When writing my own code I ended up confused and in a way I wish I did it your way instead :)